python/ray/_private/runtime_env/working_dir.py

```diff
@@ -159,14 +164,27 @@ def match(p: Path):
         return None
 
 
-def _store_package_in_gcs(gcs_key: str, data: bytes) -> int:
-    if len(data) >= GCS_STORAGE_MAX_SIZE:
+def _store_package_in_gcs(
+        gcs_key: str,
+        data: bytes,
+        logger: Optional[logging.Logger] = default_logger) -> int:
+    file_size = len(data)
+    file_size_str = f"{file_size/(1024 * 1024):.1f} MiB"
+
+    if file_size >= GCS_STORAGE_MAX_SIZE:
         raise RuntimeError(
-            "working_dir package exceeds the maximum size of 100MiB. You "
-            "can exclude large files using the 'excludes' option to the "
-            "runtime_env.")
+            f"working_dir package has size {file_size_str}, which exceeds the "
+            "maximum size of 100 MiB. You can exclude large files using the "
+            "'excludes' field of the runtime_env.")
+
+    if file_size > SILENT_UPLOAD_SIZE_THRESHOLD:
+        logger.warning(f"Pushing local files to cluster ({file_size_str})...")
```

Comment:
I'm not sure `logger.warning` is the right way to go here.  I tried `logger.info` and the following happened:

- When using Ray client, the logs didn't appear.
- When not using Ray client, the log did appear: `2021-09-24 15:51:25,101 INFO working_dir.py:428 -- Pushing local files to cluster (50.00 MiB)...`

In both cases, the `logger` kwarg isn't being passed, so it's just using `default_logger = logging.getLogger(__name__)` and I'm not sure why there's a discrepancy.

Reply:
With `logger.warning`, we get the following:
 - When using Ray client, we get a log line `Uploading local files (99.0 MiB)...`
 - When not using Ray client, we get a formatted log line: `2021-09-24 17:21:18,947 WARNING working_dir.py:181 -- Pushing local files to cluster (2.3 MiB)...`
 
We don't want that `WARNING` to show up, so I'll have to figure out what's going on here